components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/workflow/ApplicationDeletionSimpleWorkflowExecutor.java

```diff
@@ -0,0 +1,97 @@
+ */
+
+package org.wso2.carbon.apimgt.impl.workflow;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.model.Application;
+import org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO;
+import org.wso2.carbon.apimgt.impl.dto.ApplicationWorkflowDTO;
+import org.wso2.carbon.apimgt.impl.dto.WorkflowDTO;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * WS workflow executor for application delete action
+ */
+public class ApplicationDeletionSimpleWorkflowExecutor extends WorkflowExecutor {
+
+    private static final Log log = LogFactory.getLog(ApplicationDeletionSimpleWorkflowExecutor.class);
+
+    @Override
+    public String getWorkflowType() {
+        return WorkflowConstants.WF_TYPE_AM_APPLICATION_DELETION;
+    }
+
+    @Override
+    public List<WorkflowDTO> getWorkflowDetails(String workflowStatus) throws WorkflowException {
+        return null;
+    }
+
+    @Override
+    public void execute(WorkflowDTO workflowDTO) throws WorkflowException {
+        workflowDTO.setStatus(WorkflowStatus.APPROVED);
+        complete(workflowDTO);
+        super.publishEvents(workflowDTO);
+    }
+
+    @Override
+    public void complete(WorkflowDTO workflowDTO) throws WorkflowException {
+        ApiMgtDAO apiMgtDAO = new ApiMgtDAO();
+        ApplicationWorkflowDTO applicationWorkflowDTO = (ApplicationWorkflowDTO) workflowDTO;
+        Application application = applicationWorkflowDTO.getApplication();
+        Connection conn = null;
+        String errorMsg = null;
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            conn.setAutoCommit(false);
+            apiMgtDAO.deleteApplication(application, conn);
+            conn.commit();
+        } catch (APIManagementException e) {
+            if (e.getMessage() == null) {
+                errorMsg = "Couldn't complete simple application deletion workflow for application: " + application
+                        .getName();
+            } else {
+                errorMsg = e.getMessage();
+            }
+            throw new WorkflowException(errorMsg, e);
+        } catch (SQLException e) {
+            if (conn != null) {
+                try {
+                    conn.rollback();
+                } catch (SQLException ex) {
+                    log.error("Failed to rollback remove application ", ex);
+                }
+            }
+            errorMsg = "Couldn't remove application entry for application: " + application.getName();
+            throw new WorkflowException(errorMsg, e);
+        } finally {
+            try {
+                if (conn != null) {
+                    conn.close();
+                }
+            } catch (SQLException e) {
+                log.error("Couldn't close database connection of delete application workflow", e);
```

Comment:
It's better to add a comment stating the reason for just logging the exception (and continue).